cardano-api/src/Cardano/Api/Address.hs

```diff
@@ -263,6 +263,25 @@ instance SerialiseAddress (Address ShelleyAddr) where
       either (const Nothing) Just $
       deserialiseFromBech32 (AsAddress AsShelleyAddr) t
 
+instance ToJSON (Address ShelleyAddr) where
+    toJSON = Aeson.String . serialiseAddress
+
+instance ToJSON (Address ByronAddr) where
+    toJSON = Aeson.String . serialiseAddress
+
+instance FromJSON (Address ByronAddr) where
+    parseJSON = Aeson.withText "Address" $ \txt ->
+      maybe
+        (fail "Invalid Byron address.")
+        pure
+        (deserialiseAddress AsByronAddress txt)
+
+instance FromJSON (Address ShelleyAddr) where
+    parseJSON = Aeson.withText "Address" $ \txt ->
+      maybe
+        (fail "Invalid Shelley address.")
```

Comment:
I would update the error messages to "Cardano.Api.Address.FromJSON: Invalid $era address"

Reply:
OK done that